packages/tools/src/stateManagement/segmentation/config/segmentationVisibility.ts

```diff
@@ -53,26 +81,40 @@ function getSegmentationVisibility(
   toolGroupId: string,
   segmentationRepresentationUID: string
 ): boolean | undefined {
-  const toolGroupSegRepresentations =
+  const toolGroupSegmentationRepresentations =
     getSegmentationRepresentations(toolGroupId);
 
-  const segmentationData = toolGroupSegRepresentations.find(
+  const representation = toolGroupSegmentationRepresentations.find(
     (representation: ToolGroupSpecificRepresentation) =>
       representation.segmentationRepresentationUID ===
       segmentationRepresentationUID
   );
 
-  if (!segmentationData) {
+  if (!representation) {
     return;
   }
 
-  return segmentationData.visibility;
+  const { segmentsHidden } = representation;
+
+  return segmentsHidden.size === 0;
```

Comment:
I thought I read in `SegmentationStateType` that if `segmentsHidden` contains `-1` then all segments are hidden. So shouldn't that need to be accounted for here?

Reply:
I will remove it that was wrong old implementation